core/app/models/spree/stock/package.rb

```diff
@@ -108,7 +108,10 @@ def shipping_categories
       # @return [Array<Spree::ShippingMethod>] the shipping methods available
       #   for this pacakges shipping categories
       def shipping_methods
```

Comment:
this is the intersection of the variant's shipping categories' shipping methods and the stock location's shipping methods right?

i wonder if something along these lines might be a bit more expressive:
```ruby
shipping_method_sets = shipping_categories.map(&:shipping_methods)
shipping_method_sets << stock_location.shipping_methods
shipping_method_sets.reduce(:&).to_a
```